Mathlib/Tactic/Expect.lean

```diff
@@ -0,0 +1,43 @@
+/-
+Copyright (c) 2022 Arthur Paulino. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino
+-/
+
+import Lean
+
+namespace Mathlib.Tactic
+
+open Lean Meta Elab Tactic
+
+/-- `expect_failure tacs` succeeds iff `tacs` fails. -/
+elab "expect_failure " tacs:tacticSeq : tactic => withMainContext do
```

Comment:
Is `withMainContext` needed for `expect_failure`/`expect_failure_msg`/`expect_goal`? It seems maybe not; I would like to understand better when it is needed / desirable!

(It is needed for `expect_hyp`.)

Reply:
Nice catch!
I don't have a complete understanding either. Maybe @digama0 or @gebner can help us here